src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,134 @@
+    }
+
+    @BeforeEach
+     void setUp() {
+        user = new User();
+        user.setLogin(DEFAULT_LOGIN);
+        user.setAge(DEFAULT_AGE);
+        user.setPassword(DEFAULT_PASSWORD);
+    }
+
+    @Test
+    void register_correct_ok() {
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void register_maxAge_ok() {
+        user.setAge(MAX_AGE);
+        assertEquals(user, registrationService.register(user));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(NULL_STRING);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(NULL_STRING);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        user.setPassword(EMPTY_STRING);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyPlacePassword_notOk() {
+        user.setPassword(DEFAULT_PASSWORD + EMPTY_PLACE);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyPlaceLogin_notOk() {
+        user.setPassword(DEFAULT_LOGIN + EMPTY_PLACE);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(EMPTY_STRING);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        user.setPassword(SHORT_PASSWORD);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_smallAge_notOk() {
+        user.setAge(SMALL_AGE);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(NEGATIVE_AGE);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_noAge_notOk() {
+        user.setAge(NULL_AGE);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_bigAge_notOk() {
+        user.setAge(BIG_AGE);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userRegistered_notOk() {
+        User userRegistered = user;
+        registrationService.register(userRegistered);
+        assertThrows(UserNotFoundException.class, () -> registrationService.register(user));
+
+    }
 
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
 }
```

Comment:
All in all, this is how Mykyta asked me to name my tests
register_addAgeGreaterThanMinAge_ok()
register_addAdultUser_ok()
register_addUserWithAgeLessThanMinAge_NotOk()
register_addNullAge_NotOk()
register_addNullUser_NotOk()
register_addNullLogin_NotOk()
register_loginExists_NotOk()
register_passwordLengthIsGreaterThanMinLength_Ok()
register_passwordLengthIsLessThanMinLength_Ok()
register_addNullPassword_NotOk()
register_passwordLengthIsLessThanMinLength_Ok()